client/Library/Components/ListingRow.tsx

```diff
@@ -5,6 +5,13 @@ import { linkComponentToObservables } from "../../Combatant/linkComponentToObser
 import { Listing } from "../Listing";
 import { ListingButton } from "./ListingButton";
 
+export interface ExtraButton<T extends Listable> {
```

Comment:
Cool pattern. If you'd like to expand this approach to cover all of the ListingButtons, that would be a bonus, but not required for this PR.

Reply:
I've started on this, but I'll throw up a second PR so the scope of this one doesn't get too out of hand.

Reply:
For a sneak peek:
https://github.com/danlite/improved-initiative/compare/minion-boss-hp...danlite:generic-listing-buttons